tools/elfdeps.c

```diff
@@ -146,6 +244,8 @@ static void processVerDef(Elf_Scn *scn, GElf_Shdr *shdr, elfInfo *ei)
 		    continue;
 		} else if (soname && !soname_only && !skipPrivate(s)) {
 		    addDep(&ei->provides, soname, s, ei->marker);
+		    if (multiarch_deps)
+			addDep(&ei->provides, soname, s, ei->archmarker);
```

Comment:
Hmm, shouldn't the non-archmarker addDep() be behind an else, like it is in all the other similar cases?

And in that case, the multiarch_deps condition is better handled inside addDep() rather than repeating it several times in the codebase (and getting it wrong in some).

Reply:
This is for backwards compatibility, especially for supporting third party packages.